site/controllers/search.php

```diff
@@ -3,29 +3,24 @@
 use Kirby\Cms\App;
 
 return function (App $kirby) {
-	$query   = trim(get('q', ''));
-	$area    = trim(get('area', ''));
-	$results = null;
+	$query = trim(get('q', ''));
+	$area  = trim(get('area', ''));
+	$areas = $kirby->option('search.areas')($kirby);
 
-	if (empty($query) === false) {
-		$params = [
-			'hitsPerPage'           => (int)get('limit', 50),
-			'attributesToHighlight' => false,
-			'attributesToSnippet'   => '*'
-		];
-
-		if (empty($area) == false && $area !== 'all') {
-			$params['filters'] = 'area:' . $area;
-		}
-
-		$results = algolia()->query($query, param('page') ?? 1, $params);
+	if ($query !== '') {
+		$results = search(
+			area:  $area,
+			query: $query,
+			page:  param('page', 1),
+			limit: get('limit', 50)
+		);
 	}
 
 	return [
 		'results'    => $results ?? [],
```

Comment:
```suggestion
		'results'    => $results ??= [],
```

We should use like that or we define `$results` variable as `null` as we did before.

Reply:
Why? Where else are we passing `$results`?

Reply:
We are using `$result` variable in next line.

Reply:
But we also check with `?` if it exists. If we assign `[]` that would break as we cannot call a method on an array.

Reply:
AFAIK `?->` is for `null` check but `$results` is not defined if query is empty.

Reply:
Ah you are right. I thought this would work also for non-existing. I think it needs to be `$results?->pagination() ?? null` then it seems to work.